File: ios/PTimerKit/Sources/PTimerCore/Timer/ScheduledTimerTarget.swift
// Copyright © 2026 Sangwook Han
// SPDX-License-Identifier: Apache-2.0

import Foundation

/// Pure value type shared by the app's Live Activity, the widget extension,
/// and the lock-screen coordinator. Lives in PTimerCore so the widget can use
/// it without depending on the full PTimerKit layer.
public struct ScheduledTimerTarget: Codable, Equatable, Hashable {
    public let timerID: UUID
    public let timerName: String
    public let endDate: Date

    public init(timerID: UUID, timerName: String, endDate: Date) {
        self.timerID = timerID
        self.timerName = timerName
        self.endDate = endDate
    }
}

File: android/core/src/main/kotlin/com/sangwook/ptimer/core/timer/ScheduledTimerTarget.kt
// Copyright © 2026 Sangwook Han
// SPDX-License-Identifier: Apache-2.0

package com.sangwook.ptimer.core.timer

import java.time.Instant
import java.util.UUID

/**
 * Pure value type shared by the timer coordinator and the notification /
 * ongoing-timer surfaces. Port of iOS PTimerCore ScheduledTimerTarget.
 */
data class ScheduledTimerTarget(
    val timerID: UUID,
    val timerName: String,
    val endDate: Instant,
)
